scrapy/extensions/robotstxtparser.py

```diff
@@ -0,0 +1,97 @@
+from six.moves.urllib_robotparser import RobotFileParser
+
+from scrapy.utils.python import to_native_str, to_unicode
+
+class PythonRobotParser():
+    def __init__(self, content):
+        try:
+            content = to_native_str(content)
+        except UnicodeDecodeError:
+            # If we found garbage or robots.txt in an encoding other than UTF-8, disregard it.
+            # Switch to 'allow all' state.
+            content = ''
+        self.rp = RobotFileParser()
+        self.rp.parse(content.splitlines()) 
+
+    def allowed(self, url, useragent):
+        try:
+            useragent = to_native_str(useragent)
+            url = to_native_str(url)
+        except UnicodeDecodeError:
+            return False
+        return self.rp.can_fetch(useragent, url)
+
+    def sitemaps(self):
+        """RobotFileParser does not support Sitemaps directive."""
+        def _get_empty_generator():
```

Comment:
You can remove this line, the `return _get_empty_generator()` one, and unindent the remaining 2 lines, and it should work the same way :)